native/src/main/java/io/ballerina/stdlib/http/api/HttpUtil.java

```diff
@@ -575,9 +583,12 @@ public static BError createHttpError(Throwable throwable) {
                                      IOUtils.getIOPackage());
             return createHttpError("Something wrong with the connection", HttpErrorType.GENERIC_CLIENT_ERROR, cause);
         } else if (throwable instanceof ClientConnectorException) {
-            cause = createErrorCause(throwable.getMessage(), IOConstants.ErrorCode.GenericError.errorCode(),
-                                     IOUtils.getIOPackage());
-            return createHttpError("Something wrong with the connection", HttpErrorType.GENERIC_CLIENT_ERROR, cause);
+            BMap<BString, Object> detail = ValueCreator.createMapValue(MAP_TYPE);
```

Comment:
Rather than creating a custom map, we can directly create the `Detail` record here:
```suggestion
            BMap<BString, Object> detail = ValueCreator.createRecordValue(ModuleUtils.getHttpPackage(), "Detail");
```